forc-pkg/src/pkg.rs

```diff
@@ -1957,9 +1962,62 @@ pub fn compile(
         warnings,
         metrics,
     };
+
+    #[cfg(feature = "profile")]
+    report_assembly_information(&asm, &compiled_package);
+
     Ok(compiled_package)
 }
 
+#[cfg(feature = "profile")]
+fn report_assembly_information(
+    compiled_asm: &sway_core::CompiledAsm,
+    compiled_package: &CompiledPackage,
+) {
+    let mut bytes = compiled_package.bytecode.bytes.clone();
+
+    let data_offset = u64::from_be_bytes(bytes.iter().skip(8).take(8).cloned().collect::<Vec<_>>().try_into().unwrap());
+    let data_section_size = bytes.len() as u64 - data_offset;
+
+    bytes.truncate(data_offset as usize);
+
+    let mut data_section_used = 0u64;
+
+    fn calculate_entry_size(entry: &sway_core::asm_generation::Entry) -> u64 {
+        let padding = match &entry.padding {
+            sway_ir::Padding::Left { target_size } => target_size,
+            sway_ir::Padding::Right { target_size } => target_size,
+        };
+
+        let value_size = match &entry.value {
+            sway_core::asm_generation::Datum::Byte(x) => std::mem::size_of_val(x) as u64,
+            sway_core::asm_generation::Datum::Word(x) => std::mem::size_of_val(x) as u64,
+            sway_core::asm_generation::Datum::ByteArray(x) => std::mem::size_of_val(x) as u64,
+            sway_core::asm_generation::Datum::Slice(slice) => slice.len() as u64,
+            sway_core::asm_generation::Datum::Collection(entries) => entries.iter().map(calculate_entry_size).sum(),
+        };
+
+        assert_eq!(*padding as u64, value_size);
+
+        value_size
+    }
+
+    for entry in &compiled_asm.0.data_section.value_pairs {
+        data_section_used += calculate_entry_size(entry);
+    }
+
+    let asm_information = sway_core::asm_generation::AsmInformation {
+        bytecode_size: bytes.len() as _,
+        data_section: sway_core::asm_generation::DataSectionInformation { 
+            size: data_section_size, 
+            used: data_section_used, 
+            value_pairs: compiled_asm.0.data_section.value_pairs.clone(),
+        }
+    };
+
+    println!("/forc-perf info {}", serde_json::to_string(&asm_information).unwrap());
```

Comment:
> It communicates with an external forc-perf executable in order to signal the beginning and end of different compilation phases for profiling purposes.

So it communicates with forc-perf over stdout? Could you add to the PR description links to forc-perf documentation, and how to test this locally?

I'm wondering why `forc-perf` doesn't just invoke `forc_pkg::build` and have the performance information returned from the function? Or a separate wrapper function like `forc_pkg::build_with_profiler`